service-discovery/consul/src/main/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscovery.java

```diff
@@ -11,17 +9,24 @@
 import io.smallrye.stork.CachingServiceDiscovery;
 import io.smallrye.stork.DefaultServiceInstance;
 import io.smallrye.stork.ServiceInstance;
+import io.smallrye.stork.ServiceMetadata;
 import io.smallrye.stork.config.ServiceDiscoveryConfig;
 import io.smallrye.stork.spi.ServiceInstanceIds;
 import io.smallrye.stork.spi.ServiceInstanceUtils;
 import io.vertx.core.Vertx;
 import io.vertx.ext.consul.ConsulClient;
 import io.vertx.ext.consul.ConsulClientOptions;
+import io.vertx.ext.consul.Service;
 import io.vertx.ext.consul.ServiceEntry;
 import io.vertx.ext.consul.ServiceEntryList;
 
 public class ConsulServiceDiscovery extends CachingServiceDiscovery {
 
+    public static final String META_CONSUL_SERVICE_ID = "consul-service-id";
```

Comment:
The main question is how do we communicate these keys? They are kind of an API that cannot be discovered.

Reply:
I've just pushed a version with labels and metadata in the ServiceInstance, I will work on this now